ironic_cleanup.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+function wait_condition () {
+condition=0
+while [ "$condition" != "3" ] ; do
+  condition=$( openstack baremetal node list -c 'Provisioning State' | grep $1 | wc -l)
+  echo "waiting 5s for servers to be $1"
+  sleep 5
+done
+}
+
+export OS_TOKEN=fake-token
+export OS_URL=http://localhost:6385/
+
+SERVERS=`openstack baremetal node list -f value -c Name| xargs`
+if [ -z "$SERVERS" ] ; then
+    echo "nothing to do"
+    exit 0
+fi
+for server in $SERVERS ; do openstack baremetal node undeploy $server ; done
```

Comment:
as far as I like one liner, maybe you can rewrite the for loops with a single function ?
unless we want to execute an operation at a time to all servers first